src/analyze.ts

```diff
@@ -708,7 +719,9 @@ module.exports = async function (id, code, job) {
     },
     leave (node, parent) {
       if (node.scope) {
-        scope = scope.parent;
+        if (scope.parent) {
+          scope = scope.parent;
+        }
```

Comment:
Is it safe to assume that we don't want to assign the scope to `undefined` when the `scope.parent` is undefined?

Reply:
So just like `enter`, the only time `parent` is undefined is when `node` is `Program`. So yes this would assign `undefined` to scope when leaving at the very end of the run, but no other code checking `scope` would run after that.

So it's fine to either include this check or to enforce that scope is always defined since the case where it is not defined is effectively non-executable.